### ISSUE TITLE
SIMD is not working for IsInf

### ISSUE THREAD

**Issue by hamptonm1** [onnx/onnx-mlir#2279]
When running the lit test for operation IsInf using SIMD, the program crashes with the following error:

`lhs.getType() == rhs.getType() && "expected same type"` for line 387. After talking to @AlexandreEichenberger he helped me realize that the types are different we are trying to compare a scalar to a vector and that is why it fails. So we need to utilize `create.vec.splat(vecType,constant)` to ensure the conversion to vector.

However, when I attempted to do so I continued to receive an error. So I am handing this over to @AlexandreEichenberger for further investigation. 

Here is skeleton code I wrote to use `splat`:

```
  // For vectors, need to splat the constant.
  VectorType vecType = inputElemType.dyn_cast<VectorType>();
  Value negInfVec = create.vec.splat(vecType, negInf);
  Value posInfVec = create.vec.splat(vecType, posInf);
```

**Comment by hamptonm1** [comment]
Also for the IsInf operation.... I was using inputElemType opposed to elementType because for IsInf, when I use elementType it gives me the output value of `i1*` which corresponds to the below error message. It is complaining that the input is an integer and that this custom operation does not support it. This occurs because elementType is the output value which is an integer and the input value is supposed to be a float. 

```
this op does not supports custom scalar for integers
UNREACHABLE executed at /workdir/onnx-mlir/src/Conversion/ONNXToKrnl/Math/Elementwise.cpp:45!
PLEASE submit a bug report to https://github.com/llvm/llvm-project/issues/ and include the crash backtrace.
Stack dump:
0.      Program arguments: /workdir/onnx-mlir/build/Debug/bin/onnx-mlir-opt --shape-inference --convert-onnx-to-krnl /workdir/onnx-mlir/test/mlir/onnx/onnx_lowering.mlir -split-input-file
 #0 0x0000004004d25eff llvm::sys::PrintStackTrace(llvm::raw_ostream&, int) (/workdir/onnx-mlir/build/Debug/bin/onnx-mlir-opt+0x4d25eff)
 #1 0x0000004004d238f4 SignalHandler(int) Signals.cpp:0:0
 #2 0x0000004008de3420 __restore_rt (/lib/x86_64-linux-gnu/libpthread.so.0+0x14420)
 #3 0x00000040091cf00b raise /build/glibc-SzIz7B/glibc-2.31/signal/../sysdeps/unix/sysv/linux/raise.c:51:1
 #4 0x00000040091ae859 abort /build/glibc-SzIz7B/glibc-2.31/stdlib/abort.c:81:7
 #5 0x0000004004c66fae (/workdir/onnx-mlir/build/Debug/bin/onnx-mlir-opt+0x4c66fae)
 #6 0x0000004000c07fcd void onnx_mlir::CheckIfCustomScalarOpIsSupported<mlir::ONNXIsInfOp>(mlir::Type) /workdir/onnx-mlir/src/Conversion/ONNXToKrnl/Math/Elementwise.cpp:47:45
 #7 0x0000004000c03240 mlir::Value onnx_mlir::emitScalarOpFor<mlir::ONNXIsInfOp>(mlir::ConversionPatternRewriter&, mlir::Location, mlir::Operation*, mlir::Type, llvm::ArrayRef<mlir::Value>) /workdir/onnx-mlir/src/Conversion/ONNXToKrnl/Math/Elementwise.cpp:369:71
 #8 0x0000004000cca50c onnx_mlir::ONNXElementwiseUnaryOpLowering<mlir::ONNXIsInfOp>::matchAndRewrite(mlir::ONNXIsInfOp, mlir::ONNXIsInfOpAdaptor, mlir::ConversionPatternRewriter&) const::‘lambda’(onnx_mlir::KrnlBuilder&, mlir::ValueRange)::operator()(onnx_mlir::KrnlBuilder&, mlir::ValueRange) const /workdir/onnx-mlir/src/Conversion/ONNXToKrnl/Math/Elementwise.cpp:1997:71
```

**Comment by AlexandreEichenberger** [comment]
issue addressed in #2319
